app/src/main/java/com/keylesspalace/tusky/adapter/PlaceholderViewHolder.kt

```diff
@@ -14,27 +14,55 @@
  * see <http://www.gnu.org/licenses>. */
 package com.keylesspalace.tusky.adapter
 
+import android.util.Log
 import android.view.View
-import android.widget.Button
-import android.widget.ProgressBar
 import androidx.recyclerview.widget.RecyclerView
+import com.google.android.material.button.MaterialButton
+import com.google.android.material.progressindicator.CircularProgressIndicatorSpec
+import com.google.android.material.progressindicator.IndeterminateDrawable
 import com.keylesspalace.tusky.R
 import com.keylesspalace.tusky.interfaces.StatusActionListener
 
 /**
  * Placeholder for different timelines.
- * Either displays "load more" button or a progress indicator.
- **/
+ *
+ * Displays a "Load more" button for a particular status ID.
+ *
+ * If loading for the placeholder is already active the user cannot interact
```

Comment:
Question/observation about the comments in this PR: they seem rather lenghty and mostly explain what also the code should explain?

In this case here for example the class is so short that I would expect the code to speak mostly for itself.

As far as I understand comment lore they should - when necessary at all - explain the why and maybe what but normally not the details.

Having said that I do feel the need for _some_ more comments in the (existing) code to give some hint what is happening there. But a one-liner is normally sufficient for that.

Reply:
Would you prefer something like:

```
/**
 * Displays a "Load more" button for a particular status ID, or a
 * circular progress wheel if the status' page is being loaded.
 *
 * The user can only have one "Load more" operation in progress at
 * a time (determined by the adapter), so the contents of the view
 * and the enabled state is driven by that.
 */
```

Reply:
Yes, that sounds much better.